lib/src/plugin_controller.dart

```diff
@@ -119,10 +110,10 @@ class PluginController
   final Stream<List<int>> _bleStatusRawChannel;
   final Logger _debugLogger;
 
-  Stream<ConnectionStateUpdate> _connectionUpdateStream;
-  Stream<CharacteristicValue> _charValueStream;
-  Stream<ScanResult> _scanResultStream;
-  Stream<BleStatus> _bleStatusStream;
+  Stream<ConnectionStateUpdate>? _connectionUpdateStream;
+  Stream<CharacteristicValue>? _charValueStream;
+  Stream<ScanResult>? _scanResultStream;
+  Stream<BleStatus>? _bleStatusStream;
```

Comment:
Can these be just `late`?

Reply:
we do a one time initialization later on. 

```dart
 @override
  Stream<ConnectionStateUpdate> get connectionUpdateStream =>
      _connectionUpdateStream ??= _connectedDeviceRawStream
          .map(_protobufConverter.connectionStateUpdateFrom)
          .map(
        (update) {
          _debugLogger.log(
            'Received $ConnectionStateUpdate(deviceId: ${update.deviceId}, connectionState: ${update.connectionState}, failure: ${update.failure})',
          );
          return update;
        },
      );

```

With `late` keyword we would loose this since then we would get:

```dart
 Stream<ConnectionStateUpdate> get connectionUpdateStream =>
      _connectedDeviceRawStream
          .map(_protobufConverter.connectionStateUpdateFrom)
          .map(
        (update) {
          _debugLogger.log(
            'Received $ConnectionStateUpdate(deviceId: ${update.deviceId}, connectionState: ${update.connectionState}, failure: ${update.failure})',
          );
          return update;
        },
      );
```
which I think is not what we want to achieve here

Reply:
Your second example doesn't use `_connectionUpdateStream` at all, so not clear, what you mean. But any way, probably not applicable.